### PR TITLE
Cyberpunk themed clothes and armor

### PR DESCRIPTION
**Pull request by ErhardSteinhauer**
## About the PR
_**In draft until https://github.com/new-frontiers-14/frontier-station-14/pull/1329 and https://github.com/new-frontiers-14/frontier-station-14/pull/1338 merged**_

### Things added:
- More jackets and coats with somewhat (cyber)punkish/bladerunnery vibes.
- Procedurally generated punkish jumpsuits, jackets, armor vests, belts gas masks, glasses and boots. No helmets though - messes up hair and y'all know how expensive that space hair gel is out here.
- New clothes are available through loadouts system.
- Vandalized versions of ClothesMate (ThreadsChoom) and BountyVend (BonanzaMatic) filled with new clothing articles and armor.
- Space punk gangoons.
- Holograffities and special projector that allows you to put graffities on walls, windows, vendomats, terminals.
- New accent, based on slang terms from Cyberpunk 2020 TTRPG, Cyberpunk 2077, Shadowrun 5e TTRPG and Informal Contractions.

### Bundled changes:
- Moved full auto capable guns from T2 to T3+ expedition spawners

### To Do
- [x] New vendomat where one can get new clothes.
- [x] More clothes.
- [x] Graffiti holoprojector.
- [x] __**More**__ graffiti. Pushed the number to 43 in total. Should suffice for now.
- [x] Goggles.
- [x] Glowing gas masks.
- [x] Improve procedural armor visuals.
- [x] More loadout variants for gangers.
- [x] Salvage expedition faction.
- [x] Further randomize loot from expedition mobs.
- [x] Add some clothes to loadouts.

## Why / Balance
More drip

## How to test
Spawn new items, wear.
Spawn graffiti holoprojector, apply it to various surfaces.

### Clothes and armor
***Jackets***
image
***Armor and jumpsuits***
image
***Goggles, infoshades, halfmasks, webbings and boots***
image

### Graffiti

- [x] I have added screenshots/videos to this PR showcasing its changes ingame

## Breaking changes
none afaik

## Known issues
- Randomized visuals don't support icons and inhand sprites (or rather there is no way of synchronizing randomly picked colors between worn sprites and icons/inhand sprites afaik), so procedurally generated clothing articles don't come with normal icons or inhand sprites.

**Changelog**
:cl: erhardsteinhauer
- add: Added more questionable clothing choices for fashion enjoyers: jumpsuits, jackets, armor vests, glasses, gasmasks and belts. Check your loadouts.
- add: Added holograffiti projector.
- add: It is now possible to encounter bands of gangers planetside.
- add: Added new accent - Street Punk.

### CODE REVIEW

**Comment by mrscratch** [comment]
I'm digging the letterman jackets. Good stuff

**Comment by ErhardSteinhauer** [comment]
> I'm digging the letterman jackets. Good stuff

ngl, out of all things added in this PR so far, my fav is magenta letterman jacket

**Comment by ErhardSteinhauer** [comment]
Well, shit. Stupid yaml linter failed to catch the error I'm getting in console upon running this stuff.

File: RobustToolbox
Comment (ErhardSteinhauer) [inline RobustToolbox:1]: How do I remove this thing?

**Comment by ErhardSteinhauer** [comment]
> oh the glowing umbrellas from blade runner yes!

extremely useful on space station and pair well with semitransparent rain jackets

File: RobustToolbox
Comment (dvir001) [reply RobustToolbox:1]: Ignore it for now, as its going to overwrite anyway when we upstream into this.

**Comment by mrscratch** [comment]
Looks great! Some of the jumpsuits could use a shorts variation so players can show off leg tattoos. Other then that we really needed more clothing options. Good stuff!

**Comment by ErhardSteinhauer** [comment]
No longer in draft, ready to be reviewed.

**Comment by ErhardSteinhauer** [comment]
errors free too!

File: Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl
Comment (whatston3) [inline Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl:505]: As above, reads easier to me.
```suggestion
accent-streetpunk-replacement-153 = deets
accent-streetpunk-replacement-154 = deets
```

File: Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl
Comment (whatston3) [inline Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl:526]: leave->delta, great
leaves->deltas?  Diona most affected.
leaving->deltaing?
left->delta'd
```suggestion
accent-streetpunk-replacement-172 = delta
accent-streetpunk-replacement-173 = deltas
accent-streetpunk-replacement-174 = deltain'
accent-streetpunk-replacement-175 = delta'd
```

File: Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl
Comment (whatston3) [inline Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl:234]: Typo.
```suggestion
accent-streetpunk-replaced-227 = pseudonym
```

File: Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl
Comment (whatston3) [inline Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl:597]: Delta part 2?

Examples and replacements:
You going to run away from a fight -> You gonna delta from a fight
He's running away from here -> He's deltain' from here
She pulls out a gun and runs away -> She pulls out a iron (oof) and deltas.

```suggestion
accent-streetpunk-replacement-244 = delta
accent-streetpunk-replacement-245 = deltain'
accent-streetpunk-replacement-246 = deltas
```

**Review by whatston3 — COMMENTED** [review]
Accent review finished.  Looking forward to seeing this in game.

File: Resources/Locale/en-US/_NF/advertisements/mobchatter/spacepunkhumanoidmob.ftl
Comment (whatston3) [inline Resources/Locale/en-US/_NF/advertisements/mobchatter/spacepunkhumanoidmob.ftl:21]: Sorry, but I don't know what you're going for with "And if I find?" with 2.
```suggestion
advertisement-spacepunkhumanoid-1 = Man, I hate it in here!
advertisement-spacepunkhumanoid-2 = Hey! Got a ciggy?
advertisement-spacepunkhumanoid-3 = Musta been a rat. Hate those gonks!
advertisement-spacepunkhumanoid-4 = 'Ey, listen. I don't like your mug, choom!
advertisement-spacepunkhumanoid-5 = Ha-ha! Nice one, choomba!
advertisement-spacepunkhumanoid-6 = Yo, check this out!
advertisement-spacepunkhumanoid-7 = Fuck, that blunt hits hard, I'm trippin'.
advertisement-spacepunkhumanoid-8 = Hell.. I need to get a dose. And fast.
advertisement-spacepunkhumanoid-9 = Ah, yes, s'all coming together now!
advertisement-spacepunkhumanoid-10 = Sometimes I dream about cheese...
advertisement-spacepunkhumanoid-11 = Stop!
advertisement-spacepunkhumanoid-12 = Give me ya money! And ya shoes!
advertisement-spacepunkhumanoid-13 = Best shit ever!!
advertisement-spacepunkhumanoid-14 = Why'd ya gotta make it harder?
advertisement-spacepunkhumanoid-15 = Argh!
advertisement-spacepunkhumanoid-16 = Huh, that's funny.
advertisement-spacepunkhumanoid-17 = This day's turnin' out a'ight after all!
advertisement-spacepunkhumanoid-18 = Hah! Skill issue!
advertisement-spacepunkhumanoid-19 = *whistles*
advertisement-spacepunkhumanoid-20 = Dibs on that!
advertisement-spacepunkhumanoid-21 = Should've delta'd while you still had a chance, gonk!
```

File: Resources/Locale/en-US/_NF/advertisements/vending/vandalizedbountyvend.ftl
Comment (whatston3) [inline Resources/Locale/en-US/_NF/advertisements/vending/vandalizedbountyvend.ftl:7]:
```suggestion
﻿advertisement-vandalizedbountyvend-1 = Get some iron!
advertisement-vandalizedbountyvend-2 = Iron! Breakdowns! GUNS BONANZA!
advertisement-vandalizedbountyvend-3 = You packin', choom?
advertisement-vandalizedbountyvend-4 = Start zeroin' 'ganic gonks left an' right with this iron!
advertisement-vandalizedbountyvend-5 = Badges don't have to know the deets.
advertisement-vandalizedbountyvend-6 = Prep your scratch, choom! Only preem guns here!
advertisement-vandalizedbountyvend-7 = I'm only barfin' nova iron!
```

File: Resources/Locale/en-US/_NF/advertisements/vending/vandalizedclothesmate.ftl
Comment (whatston3) [inline Resources/Locale/en-US/_NF/advertisements/vending/vandalizedclothesmate.ftl:7]: Reminds me - should the accent have your->ya?

Suggesting preem->nova once because it's used a lot here.
```suggestion
﻿advertisement-vandalizedclothes-1 = I don't dig ya looks, choom! Get preem threads here!
advertisement-vandalizedclothes-2 = Don'tcha wanna look totally nova?
advertisement-vandalizedclothes-3 = D'you wanna look preem or what?! Shlep over here!
advertisement-vandalizedclothes-4 = Your new look will flatline your input!
advertisement-vandalizedclothes-5 = Now with added composite armor plates!
advertisement-vandalizedclothes-6 = Preem threads! Nova getups! Check 'em out, choom!
advertisement-vandalizedclothes-7 = Nova outfit you have goin' there! For A GONK!
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:7]:
```suggestion
  description: Mostly transparent raincoat made out of acid resistant polymers. Please, for the love of Space God, do wear undergarments.
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:73]:
```suggestion
  description: Ya gotta 'ave style on dem streets.
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:84]:
```suggestion
  description: Rock'n'roll, babe!
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:95]:
```suggestion
  description: Made out of toughest synthleather for toughest badasses.
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:106]:
```suggestion
  description: Welcome to jamrock.
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:117]:
```suggestion
  description: When you wear it, you can always hear the sickest guitar solo in your mind.
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:128]:
```suggestion
  description: Where is your lacrosse stick?
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:139]:
```suggestion
  description: Where is your lacrosse stick?
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:150]:
```suggestion
  description: Where is your lacrosse stick?
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:161]: Don't know what this is but I want it.
```suggestion
  description: Where is your holocrosse joystick?
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:172]:
```suggestion
  description: Oi! Whatcha lookin' at?
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:184]:
```suggestion
  description: Oi! Whatcha lookin' at?
```

File: Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/OuterClothing/coats_punk.yml:232]:
```suggestion
  description: Where is your holocrosse joystick?
```

File: Resources/Prototypes/_NF/Entities/Clothing/Shoes/boots_punk.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Clothing/Shoes/boots_punk.yml:6]:
```suggestion
  description: Deez boots were made for stompin', an' stompin' on ya bucket dey'll do.
```

**Review by whatston3 — COMMENTED** [review]
More nitpicky name/description edits.

**Comment by whatston3** [comment]
Just writing this down so I don't forget - I think it'd be nice if the holograffiti worked more like the other holoprojectors and didn't highlight the hologram on hover.  Not sure what does that off the top of my head.

**Comment by whatston3** [comment]
Gangers that spawn with a glaive don't seem to use it in combat - might be trying to default to the gun component part of it, but can't wield it.

**Comment by whatston3** [comment]
I like the design of the new clothes.  The gas mask doesn't seem to be removed from your face when you pull it up, is that intentional?

File: Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl
Comment (whatston3) [inline Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl:505]: I owe you an apology, you were right.  Looks like "deeties" to me, but oh well.
```suggestion
accent-streetpunk-replacement-153 = detes
accent-streetpunk-replacement-154 = detes
```

File: Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl
Comment (whatston3) [inline Resources/Locale/en-US/_NF/accent/streetpunk_accent.ftl:422]: I owe you an apology, I was wrong.
```suggestion
accent-streetpunk-replacement-70 = dete
accent-streetpunk-replacement-71 = detes
```

File: Resources/Prototypes/_NF/Entities/Objects/Devices/holoprojectors.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Objects/Devices/holoprojectors.yml:32]: Could use a rewrite, here's a suggestion, but I'm out of my element.
```suggestion
  description: Tag up the sharp end, let 'em know you were here. 
```

File: Resources/Prototypes/_NF/Entities/Objects/Tools/umbrella.yml
Comment (whatston3) [inline Resources/Prototypes/_NF/Entities/Objects/Tools/umbrella.yml:5]: Inherited from the lantern - suggested rewrite.
```suggestion
  description: Keeps you dry AND stylish.
```

**Review by whatston3 — COMMENTED** [review]
Finished a first pass on this, looking good so far - a few minor things.  I found a few small issues, and I was wrong on "deets", sorry about that.

I'd also say that picking the goggles/belt up when they get put on the ground is a little annoying, but not awful.

Might look into how the other holoprojectors work and suggest something for the signs getting that interaction highlight.

**Comment by ErhardSteinhauer** [comment]
> Just writing this down so I don't forget - I think it'd be nice if the holograffiti worked more like the other holoprojectors and didn't highlight the hologram on hover. Edit: it's the InteractionOutlineComponent.
> 
> Not sure if it's apparent that it gets destroyed on damage just by looking at it, but the description helps there.

I thought that outline makes it more apparent, that graffiti can be targeted. That's why I kept the component from it's parent. But I agree, that without outline it looks better.

I'll remove it. I trust players will learn how to remove graffities.

**Comment by ErhardSteinhauer** [comment]
> Gangers that spawn with a glaive don't seem to use it in combat - might be trying to default to the gun component part of it, but can't wield it. (Edit: Looking through the spawner, I don't know how they got a glaive, but I've seen a few with them - maybe they picked it up off the floor? Odd.)

I did remove glaive from their "on spawn" weapon pool, but because AI drops any items they hold in hand when they spawn (don't know why it happens) and because their weapon in hands is a spawner (you can't hold spawner in hand - it disappears on initiation and the item from spawner appears on the same tile), they go for the "best" melee option in their vicinity which isn't necessarily the one they were spawned with.

> Also, Ganger leaders spawn with both a melee weapon and a ranged weapon, but drop the melee weapon pretty much immediately - seems odd.

Same here. It doesn't drop the "gun" because there is no gun in his hands - gun is part of NPC's glove sprite.

File: Resources/Prototypes/_NF/Entities/Structures/Machines/vending_machines.yml
Comment (ErhardSteinhauer) [reply Resources/Prototypes/_NF/Entities/Structures/Machines/vending_machines.yml:524]: I think I fixed it, will push the fix.

**Comment by ErhardSteinhauer** [comment]
> I like the design of the new clothes. The gas mask doesn't seem to be removed from your face when you pull it up, is that intentional?

It is not: the mask inherits from the parent that allows it to be pullable, but it just doesn't work. No clue why.

**Comment by ErhardSteinhauer** [comment]
> I'd also say that picking the goggles/belt up when they get put on the ground is a little annoying, but not awful.

It is, yeah. I probably can add generic icons for glasses/belt, but I don't think there is a way to sync colors between sprites.

**Comment by ErhardSteinhauer** [comment]
> > I'd also say that picking the goggles/belt up when they get put on the ground is a little annoying, but not awful.
> 
> It is, yeah. I probably can add generic icons for glasses/belt, but I don't think there is a way to sync colors between sprites.

But I don't know really. The problem with generic icons in their current state that those wont represent how the item will look on the character, which imo when it comes to drip is more important than anything.

**Comment by ErhardSteinhauer** [comment]
> > I'd also say that picking the goggles/belt up when they get put on the ground is a little annoying, but not awful.
> 
> It is, yeah. I probably can add generic icons for glasses/belt, but I don't think there is a way to sync colors between sprites.

Right, I got the idea: I can increase the scale of the icon sprite. That way it'll become more visible while retaining the colors.

**Comment by ErhardSteinhauer** [comment]
> > > I'd also say that picking the goggles/belt up when they get put on the ground is a little annoying, but not awful.
> > 
> > 
> > It is, yeah. I probably can add generic icons for glasses/belt, but I don't think there is a way to sync colors between sprites.
> 
> Right, I got the idea: I can increase the scale of the icon sprite. That way it'll become more visible while retaining the colors.

A bit junky, but easier to see and interact I believe:
image

**Comment by ErhardSteinhauer** [comment]
@whatston3 biggest thanks for proofreading, I appreciate that a lot

**Comment by whatston3** [comment]
> It is, yeah. I probably can add generic icons for glasses/belt, but I don't think there is a way to sync colors between sprites.

Just as an idea, could the GenericVisualizer be used here?  I know mail edits the layers based off of the state of a mail package, but I haven't played around with it.  If you were to create a component with the colours/visible layers of your clothing and then pull that into the visualizer somehow, split it between the icon and sprite, that'd be neat.

Not suggesting to put that in this PR, but it might be interesting.

Retesting, thanks for the changes.

File: Resources/Prototypes/ai_factions.yml
Comment (whatston3) [inline Resources/Prototypes/ai_factions.yml:123]: Revs should still find dinos hostile, right?  Looking above, still true the other way around.

**Review by whatston3 — COMMENTED** [review]
Left a few small suggestions (two descriptions and a possibly unintended change to factions), but this looks good, happy to approve once you take a look at those suggestions.

Tried to test the change to NavSmash for the hostile mob base, didn't really notice much smashing, though I was checking with punks - maybe if they had better weapons they'd be more inclined to smash things to come towards me - unsure.

Still excited to see these items used in-game.  Hope the holograffiti's straightforward enough to remove.

File: Resources/Prototypes/ai_factions.yml
Comment (ErhardSteinhauer) [reply Resources/Prototypes/ai_factions.yml:123]: I probably made a mistake when I was resolving merge conflicts. Gotta take a careful look

Comment (ErhardSteinhauer) [reply Resources/Prototypes/ai_factions.yml:123]: And while I'm at it, might as well add factions for the rest on newly added expedition enemies

Comment (ErhardSteinhauer) [reply Resources/Prototypes/ai_factions.yml]: Pushed fix. Hopefully I made the ai_faction files more readable

**Review by whatston3 — APPROVED** [review]
Faction changes seem good, liking the consistent order.  Approving this, would like a second opinion given the amount of changes here.

**Pull request merged**
